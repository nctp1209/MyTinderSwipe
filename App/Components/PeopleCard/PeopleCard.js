import React, { Component } from "react";
import { View, Text, Image, StyleSheet } from "react-native";
import PropTypes from "prop-types";
import FeatherIcons from "react-native-vector-icons/Feather";

const USER = "USER";
const CALENDER = "CALENDER";
const MAP = "MAP";
const PHONE = "PHONE";
const PASSWORD = "PASSWORD";

const UNSELECTED_INFO_COLOR = "#DADADA";
const SELECTED_INFO_COLOR = "#83B943";

export default class PeopleCard extends Component {
  constructor() {
    super();
    this.state = {
      selectedInfo: MAP
    };
  }

  static propTypes = {
    item: PropTypes.object
  };

  static defaultProps = { item: {} };

  setSelectedInfo(info) {
    this.setState({ selectedInfo: info });
  }

  iconItem(iconName, iconFeature) {
    let selected = false;
    if (this.state.selectedInfo === iconFeature) selected = true;
    return (
      <View
        style={
          selected
            ? styles.buttonContainerSelectedStyle
            : styles.buttonContainerUnselectedStyle
        }
      >
        <FeatherIcons
          name={iconName}
          color={selected ? SELECTED_INFO_COLOR : UNSELECTED_INFO_COLOR}
          size={30}
          style={{ padding: 7 }}
          onPress={() => this.setSelectedInfo(iconFeature)}
        />
      </View>
    );
  }

  infoDisplay() {
    let titleDisplayed = "My address is";
    let infoDisplayed =
      this.props.item.location && this.props.item.location.street
        ? this.props.item.location.street
        : "";
    switch (this.state.selectedInfo) {
      case USER:
        titleDisplayed = "Hi, my name is";
        // Capitalize first letter of First Name
        const firstName =
          this.props.item.name && this.props.item.name.first
            ? this.props.item.name.first.charAt(0).toUpperCase() +
              this.props.item.name.first.slice(1)
            : "";

        // Capitalize first letter of Last Name
        const lastName =
          this.props.item.name && this.props.item.name.last
            ? this.props.item.name.last.charAt(0).toUpperCase() +
              this.props.item.name.last.slice(1)
            : "";

        infoDisplayed = `${firstName} ${lastName}`;
        break;
      case CALENDER:
        titleDisplayed = "My birthday is";
        infoDisplayed = this.props.item.dob
          ? new Date(this.props.item.dob * 1000).toLocaleDateString("vn-VN")
          : "";
        break;
      case PHONE:
        titleDisplayed = "My phone number is";
        infoDisplayed = this.props.item.phone ? this.props.item.phone : "";
        break;
      case PASSWORD:
        titleDisplayed = "My password is";
        infoDisplayed = this.props.item.password
          ? this.props.item.password
          : "";
        break;
    }

    return (
      <View style={styles.textContainerStyle}>
        <Text style={styles.desriptionTextStyle}>{titleDisplayed}</Text>
        <Text style={styles.textStyle}>{infoDisplayed}</Text>
      </View>
    );
  }

  render() {
    return (
      <View style={styles.containerStyle}>
        <View style={styles.backgroundStyle} />
        <View style={styles.avatarContainerStyle}>
          {this.props.item && this.props.item.picture ? (
            <Image
              style={styles.avatarStyle}
              source={{ uri: this.props.item.picture }}
            />
          ) : null}
        </View>
        {this.infoDisplay()}
        <View style={styles.buttonContainerStyle}>
          {this.iconItem("user", USER)}
          {this.iconItem("calendar", CALENDER)}
          {this.iconItem("map", MAP)}
          {this.iconItem("phone", PHONE)}
          {this.iconItem("lock", PASSWORD)}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  containerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  avatarContainerStyle: {
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "white",
    marginTop: 30,
    padding: 10,
    height: 160,
    width: 160,
    borderRadius: 160 / 2,
    borderWidth: 1,
    borderColor: "gray"
  },
  avatarStyle: {
    height: 150,
    width: 150,
    borderRadius: 150 / 2
  },
  desriptionTextStyle: {
    fontSize: 15,
    padding: 5,
    color: "gray"
  },
  textStyle: {
    fontSize: 19,
    color: "black"
  },
  textContainerStyle: {
    padding: 10,
    justifyContent: "center",
    alignItems: "center"
  },
  buttonContainerStyle: {
    flexDirection: "row",
    justifyContent: "center",
    paddingHorizontal: 5,
    paddingVertical: 30
  },
  buttonContainerSelectedStyle: {
    paddingTop: 5,
    borderTopWidth: 2,
    borderTopColor: SELECTED_INFO_COLOR
  },
  buttonContainerUnselectedStyle: {
    paddingTop: 5
  },
  backgroundStyle: {
    position: "absolute",
    top: 0,
    zIndex: -1,
    width: "100%",
    height: 140,
    backgroundColor: "#EEE",
    borderBottomColor: "gray",
    borderBottomWidth: 1,
    borderTopRightRadius: 5,
    borderTopLeftRadius: 5
  }
});
