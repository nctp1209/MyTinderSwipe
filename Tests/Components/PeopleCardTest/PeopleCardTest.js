import React from "react";
import renderer from "react-test-renderer";
import { path } from "ramda";

import APIResult from "../../APIResult";
import PeopleCard from "../../../App/Components/PeopleCard/PeopleCard";

test("PeopleCard render when no data", () => {
  const tree = renderer.create(<PeopleCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("PeopleCard render", () => {
  const response = APIResult.getPeople();
  const item = path(["data", "results"], response)[0];
  const tree = renderer.create(<PeopleCard item={item.user} />).toJSON();
  expect(tree).toMatchSnapshot();
});
