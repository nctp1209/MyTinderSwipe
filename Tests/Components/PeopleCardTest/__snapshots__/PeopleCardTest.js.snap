// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCard render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#EEE",
        "borderBottomColor": "gray",
        "borderBottomWidth": 1,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "height": 140,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": -1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 160,
        "justifyContent": "center",
        "marginTop": 30,
        "padding": 10,
        "width": 160,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://api.randomuser.me/portraits/men/11.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 75,
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 15,
          "padding": 5,
        }
      }
    >
      My address is
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 19,
        }
      }
    >
      4797 greenhaven ln
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 5,
        "paddingVertical": 30,
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#DADADA",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#DADADA",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "borderTopColor": "#83B943",
          "borderTopWidth": 2,
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#83B943",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#DADADA",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#DADADA",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`PeopleCard render when no data 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#EEE",
        "borderBottomColor": "gray",
        "borderBottomWidth": 1,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "height": 140,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": -1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderRadius": 80,
        "borderWidth": 1,
        "height": 160,
        "justifyContent": "center",
        "marginTop": 30,
        "padding": 10,
        "width": 160,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 15,
          "padding": 5,
        }
      }
    >
      My address is
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 19,
        }
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 5,
        "paddingVertical": 30,
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#DADADA",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#DADADA",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "borderTopColor": "#83B943",
          "borderTopWidth": 2,
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#83B943",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#DADADA",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#DADADA",
              "fontSize": 30,
            },
            Object {
              "padding": 7,
            },
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
